[Issue]: compiler warnings with cuda component source

### Issue Description

When compiling PAPI with the cuda component configured-in, I encounter the following warnings:

**I. papi_cupti_common.c:**
> gcc -DPAPI_NO_MEMORY_MANAGEMENT -DSTATIC_PAPI_EVENTS_TABLE  -DUSE_PERFEVENT_RDPMC=1 -DPEINCLUDE=\"libpfm4/include/perfmon/perf_event.h\" -D_REENTRANT -D_GNU_SOURCE -DUSE_COMPILER_TLS -Ilibpfm4/include -fvisibility=hidden -I. -DHAVE_CUDA -DHAVE_NVML -DPAPI_NUM_COMP=4 -I/storage/packages/cuda/12.9/include -I/storage/packages/cuda/12.9/extras/CUPTI/include -g -DPAPI_CUDA_MAIN= -DPAPI_CUDA_RUNTIME= -DPAPI_CUDA_CUPTI= -DPAPI_CUDA_PERFWORKS= -I/opt/amdgpu/include -I/storage/packages/cuda/12.9/include -DOSLOCK=\"linux-lock.h\" -DOSCONTEXT=\"linux-context.h\" -O2 -c components/cuda/papi_cupti_common.c -o papi_cupti_common.o
> components/cuda/papi_cupti_common.c: In function ‘verify_driver_branch_supports_legacy_apis’:
> components/cuda/papi_cupti_common.c:964:176: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘long int’ [-Wformat=]
>   964 | (errorMessage), "The Legacy APIs (Event and Metric) are not supported with driver branchs >= 580 (detected driver branch is %d)."
>       |                                                                                                                             ~^
>       |                                                                                                                              |
>       |                                                                                                                              int
>       |                                                                                                                             %ld
>   965 | PIs you must use a driver branch <= 575.", driverBranchDecimal);
>       |                                            ~~~~~~~~~~~~~~~~~~~                                                                
>       |                                            |
>       |                                            long int

**II. concurrent_profiling.cu:**
> /storage/packages/cuda/12.9/bin/nvcc -I. -I../../.. -I../../../testlib -I../../../validation_tests -I/storage/packages/cuda/12.9/include -I/storage/packages/cuda/12.9/extras/CUPTI/include -g -ccbin='gcc' -DPAPI     -Wno-deprecated-gpu-targets -Xcompiler -fpic  -Xcompiler -fopenmp -c -o concurrent_profiling.o concurrent_profiling.cu
> concurrent_profiling.cu: In function ‘void add_cuda_native_events_concurrent(int, std::string, perDeviceData&, int*)’:
> concurrent_profiling.cu:131:17: warning: format ‘%s’ expects argument of type ‘char*’, but argument 3 has type ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’} [-Wformat=]
>   131 |            fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name, papi_errno);
>       |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> concurrent_profiling.cu: In function ‘int main(int, char**)’:
> concurrent_profiling.cu:413:50: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘long int’ [-Wformat=]
>   413 |     PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
>       |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~ 
>       |                                                                                                                  |
>       |                                                                                                                  long int
> concurrent_profiling.cu:503:50: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
>   503 |         PRINT(global_suppress_output, "\nMetrics for device #%d:\n", i);
>       |                                                  ^~~~~~~~~~~~~~~~~~~~~~~         
>       |                                                                                 |
>       |                                                                                 size_t {aka long unsigned int}
 
**III. concurrent_profiling_noCuCtx.cu:**
> /storage/packages/cuda/12.9/bin/nvcc -I. -I../../.. -I../../../testlib -I../../../validation_tests -I/storage/packages/cuda/12.9/include -I/storage/packages/cuda/12.9/extras/CUPTI/include -g -ccbin='gcc' -DPAPI     -Wno-deprecated-gpu-targets -Xcompiler -fpic  -Xcompiler -fopenmp -c -o concurrent_profiling_noCuCtx.o concurrent_profiling_noCuCtx.cu
> concurrent_profiling_noCuCtx.cu: In function ‘void add_cuda_native_events_concurrent(int, std::string, perDeviceData&, int*)’:
> concurrent_profiling_noCuCtx.cu:130:17: warning: format ‘%s’ expects argument of type ‘char*’, but argument 3 has type ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’} [-Wformat=]
>   130 |            fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name, papi_errno);
>       |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> concurrent_profiling_noCuCtx.cu: In function ‘int main(int, char**)’:
> concurrent_profiling_noCuCtx.cu:402:50: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘long int’ [-Wformat=]
>   402 |     PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
>       |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~ 
>       |                                                                                                                  |
>       |                                                                                                                  long int
> concurrent_profiling_noCuCtx.cu:492:50: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
>   492 |         PRINT(global_suppress_output, "\nMetrics for device #%d:\n", i);
>       |                                                  ^~~~~~~~~~~~~~~~~~~~~~~         
>       |                                                                                 |
>       |                                                                                 size_t {aka long unsigned int}

### Operating System

Ubuntu 24.04.3 LTS (Noble Numbat)

### Architecture

x86_64

### CPU Architecture

AMD Ryzen 9 9950X 16-Core Processor

### Compiler

GCC 13.3.0

### PAPI Version

master

### Component

cuda

### Steps to Reproduce

1. ./configure --prefix=${INSTDIR} --with-components="cuda"
2. make
3. make install

### Additional Information

_No response_

--- a/src/components/cuda/papi_cupti_common.c
+++ b/src/components/cuda/papi_cupti_common.c
@@ -961,7 +961,7 @@ int verify_driver_branch_supports_legacy_apis(sys_compute_capabilities_e ccs_on_
     long int driverBranchDecimal = strtol(driverBranch, NULL, base);
     if (driverBranchDecimal >= 580) {
         char errorMessage[PAPI_HUGE_STR_LEN] = { 0 };
-        int strLen = snprintf(errorMessage, sizeof(errorMessage), "The Legacy APIs (Event and Metric) are not supported with driver branchs >= 580 (detected driver branch is %d)."
+        int strLen = snprintf(errorMessage, sizeof(errorMessage), "The Legacy APIs (Event and Metric) are not supported with driver branchs >= 580 (detected driver branch is %ld)."
                              " To use the Legacy APIs you must use a driver branch <= 575.", driverBranchDecimal);
         if (strLen < 0 || (size_t) strLen >= sizeof(errorMessage)) {
             SUBDBG("Failed to fully store string into buffer. Proceeding, but the sring has been truncated.\n");

--- a/src/components/cuda/tests/concurrent_profiling.cu
+++ b/src/components/cuda/tests/concurrent_profiling.cu
@@ -128,7 +128,7 @@ static void add_cuda_native_events_concurrent(int EventSet, string cuda_native_e
    int papi_errno = PAPI_add_named_event(EventSet, cuda_native_event_name.c_str());
    if (papi_errno != PAPI_OK) {
        if (papi_errno != PAPI_EMULPASS) {
-           fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name, papi_errno);
+           fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name.c_str(), papi_errno);
            exit(EXIT_FAILURE);
        }
        
@@ -404,13 +404,13 @@ int main(int argc, char **argv)
     profileKernels(device_data[0], base_cuda_native_event_names_with_stat_qual, "single_device_serial", true);
 
     auto end_time = ::std::chrono::high_resolution_clock::now();
-    auto elapsed_serial_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
+    long elapsed_serial_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
     int numBlocks = 0;
     for (i = 1; i <= numKernels; i++)
     {
         numBlocks += i;
     }
-    PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
+    PRINT(global_suppress_output, "It took %ld ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
     // Second version - same kernel calls as before on the same device, but now using separate streams for concurrency, //
@@ -424,8 +424,8 @@ int main(int argc, char **argv)
     profileKernels(device_data[0], base_cuda_native_event_names_with_stat_qual, "single_device_async", false);
 
     end_time = ::std::chrono::high_resolution_clock::now();
-    int elapsed_single_device_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
-    PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels on a single device on separate streams.\n", elapsed_single_device_ms, numKernels);
+    long elapsed_single_device_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
+    PRINT(global_suppress_output, "It took %ld ms on the host to profile %d kernels on a single device on separate streams.\n", elapsed_single_device_ms, numKernels);
     PRINT(global_suppress_output, "--> If the separate stream wallclock time is less than the serial version, the streams were profiling concurrently.\n");
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
@@ -500,7 +500,7 @@ int main(int argc, char **argv)
 
     for (i = 1; i < num_devices; i++)
     {
-        PRINT(global_suppress_output, "\nMetrics for device #%d:\n", i);
+        PRINT(global_suppress_output, "\nMetrics for device #%zu:\n", i);
         print_measured_values(device_data[i]);
     }
 

--- a/src/components/cuda/tests/concurrent_profiling_noCuCtx.cu
+++ b/src/components/cuda/tests/concurrent_profiling_noCuCtx.cu
@@ -127,7 +127,7 @@ static void add_cuda_native_events_concurrent(int EventSet, string cuda_native_e
    int papi_errno = PAPI_add_named_event(EventSet, cuda_native_event_name.c_str());
    if (papi_errno != PAPI_OK) {
        if (papi_errno != PAPI_EMULPASS) {
-           fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name, papi_errno);
+           fprintf(stderr, "Unable to add event %s to the EventSet with error code %d.\n", cuda_native_event_name.c_str(), papi_errno);
            exit(EXIT_FAILURE);
        }   
     
@@ -393,13 +393,13 @@ int main(int argc, char **argv)
     profileKernels(device_data[0], base_cuda_native_event_names_with_stat_qual, "single_device_serial", true);
 
     auto end_time = ::std::chrono::high_resolution_clock::now();
-    auto elapsed_serial_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
+    long elapsed_serial_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
     int numBlocks = 0;
     for (i = 1; i <= numKernels; i++)
     {   
         numBlocks += i;
     }   
-    PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
+    PRINT(global_suppress_output, "It took %ld ms on the host to profile %d kernels in serial.\n", elapsed_serial_ms, numKernels);
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
     // Second version - same kernel calls as before on the same device, but now using separate streams for concurrency, //
@@ -413,8 +413,8 @@ int main(int argc, char **argv)
     profileKernels(device_data[0], base_cuda_native_event_names_with_stat_qual, "single_device_async", false);
 
     end_time = ::std::chrono::high_resolution_clock::now();
-    int elapsed_single_device_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
-    PRINT(global_suppress_output, "It took %d ms on the host to profile %d kernels on a single device on separate streams.\n", elapsed_single_device_ms, numKernels);
+    long elapsed_single_device_ms = ::std::chrono::duration_cast<::std::chrono::milliseconds>(end_time - begin_time).count();
+    PRINT(global_suppress_output, "It took %ld ms on the host to profile %d kernels on a single device on separate streams.\n", elapsed_single_device_ms, numKernels);
     PRINT(global_suppress_output, "--> If the separate stream wallclock time is less than the serial version, the streams were profiling concurrently.\n");
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
@@ -489,7 +489,7 @@ int main(int argc, char **argv)
 
     for (i = 1; i < num_devices; i++)
     {   
-        PRINT(global_suppress_output, "\nMetrics for device #%d:\n", i); 
+        PRINT(global_suppress_output, "\nMetrics for device #%zu:\n", i);
         print_measured_values(device_data[i]);
     }   
 

